llama-index-integrations/readers/llama-index-readers-gcs/llama_index/readers/gcs/base.py

```diff
@@ -0,0 +1,122 @@
+        True by default.
+    file_extractor (Optional[Dict[str, BaseReader]]): A mapping of file
+        extension to a BaseReader class that specifies how to convert that file
+        to text. See `SimpleDirectoryReader` for more details.
+    required_exts (Optional[List[str]]): List of required extensions.
+        Default is None.
+    num_files_limit (Optional[int]): Maximum number of files to read.
+        Default is None.
+    file_metadata (Optional[Callable[str, Dict]]): A function that takes
+        in a filename and returns a Dict of metadata for the Document.
+        Default is None.
+    service_account_key_json (Optional[str]): provide GCP service account key JSON directly.
+    """
+
+    is_remote: bool = True
+
+    bucket: str
+    key: Optional[str] = None
+    prefix: Optional[str] = ""
+    recursive: bool = True
+    file_extractor: Optional[Dict[str, Union[str, BaseReader]]] = Field(
+        default=None, exclude=True
+    )
+    required_exts: Optional[List[str]] = None
+    filename_as_id: bool = True
+    num_files_limit: Optional[int] = None
+    file_metadata: Optional[Callable[[str], Dict]] = Field(default=None, exclude=True)
+    service_account_key_json: Optional[str] = None
+    custom_reader_path: Optional[str] = None
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "GCSReader"
+
+    def load_gcs_files_as_docs(self, temp_dir=None) -> List[Document]:
+        """Load file(s) from GCS."""
+        from gcsfs import GCSFileSystem
+
+        if self.service_account_key_json is not None:
+            creds = service_account.Credentials.from_service_account_info(
+                self.service_account_key_json, scopes=SCOPES
+            )
+        else:
+            # Use anonymous access if none are specified
+            creds = "anon"
+
+        gcsfs = GCSFileSystem(
+            token=creds,
+        )
+
+        input_dir = self.bucket
+        input_files = None
+
+        if self.key:
+            input_files = [f"{self.bucket}/{self.key}"]
+        elif self.prefix:
+            input_dir = f"{input_dir}/{self.prefix}"
+
+        loader = SimpleDirectoryReader(
+            input_dir=input_dir,
+            input_files=input_files,
+            file_extractor=self.file_extractor,
+            required_exts=self.required_exts,
+            filename_as_id=self.filename_as_id,
+            num_files_limit=self.num_files_limit,
+            file_metadata=self.file_metadata,
+            recursive=self.recursive,
+            fs=gcsfs,
+        )
+
+        return loader.load_data()
+
+    def load_data(self, custom_temp_subdir: str = None) -> List[Document]:
+        """
+        Load the file(s) from GCS.
+
+        Args:
+            custom_temp_subdir (str, optional): This parameter is deprecated and unused. Defaults to None.
+
+        Returns:
+            List[Document]: A list of documents loaded from GCS.
+        """
+        if custom_temp_subdir is not None:
```

Comment:
If this is a new package, we don't need to include unused parameters?

Reply:
🤦